src/Layouts/Partials/LayersPanel.vala

```diff
@@ -49,17 +52,41 @@ public class Akira.Layouts.Partials.LayersPanel : Gtk.ListBox {
         get_style_context ().add_class ("layers-panel");
         expand = true;
 
-        artboard = new Akira.Layouts.Partials.Artboard (window, "Artboard 1");
-        var placeholder = new Gtk.ListBoxRow ();
-        artboard.container.insert (placeholder, 0);
-        placeholder.visible = false;
-        placeholder.no_show_all = true;
+        list_model = new Akira.Models.ListModel ();
+        item_model_map = new Gee.HashMap<string, Akira.Models.LayerModel> ();
 
-        insert (artboard, 0);
+        bind_model (list_model, item => {
+            // TODO: Differentiate between layer and artboard
+            // based upon item "type" of some sort
+            return new Akira.Layouts.Partials.Layer (
+                window,
+                (Akira.Models.LayerModel) item
+            );
+        });
 
         build_drag_and_drop ();
-
         reload_zebra ();
+
+        window.event_bus.item_inserted.connect (on_item_inserted);
+        window.event_bus.item_deleted.connect (on_item_deleted);
+    }
+
+    private void on_item_inserted (Lib.Models.CanvasItem new_item) {
+        var model = new Akira.Models.LayerModel (new_item, list_model);
+        list_model.add_item.begin (model);
+
+        // This map is necessary for easily knowing which
+        // item is related to which model, since the canvas knows only
+        // real items and the layers panel only knows items model
+        item_model_map.@set (new_item.id, model);
+
+        show_all ();
+    }
+
+    private void on_item_deleted (Lib.Models.CanvasItem item_to_delete) {
+        var model_to_delete = item_model_map.@get (item_to_delete.id);
+
+        list_model.remove_item.begin (model_to_delete);
```

Comment:
Let's keep the variable names as simple and consistent as possible.
We can call those simply `item` and `model` as we don't need to be that specific inside an already eloquent method.

Reply:
Sure, I will do the same if it happens again!